x/authenticator/authenticator/any_of.go

```diff
@@ -73,38 +73,43 @@ func (aoa AnyOfAuthenticator) Initialize(data []byte) (iface.Authenticator, erro
 	return aoa, nil
 }
 
-func (aoa AnyOfAuthenticator) Authenticate(ctx sdk.Context, request iface.AuthenticationRequest) iface.AuthenticationResult {
+func (aoa AnyOfAuthenticator) Authenticate(ctx sdk.Context, request iface.AuthenticationRequest) error {
 	baseId := request.AuthenticatorId
+
+	var err error
 	for id, auth := range aoa.SubAuthenticators {
 		request.AuthenticatorId = baseId + "." + strconv.Itoa(id)
-		result := auth.Authenticate(ctx, request)
-		if result.IsAuthenticated() || result.IsRejected() {
-			return result
+		err = auth.Authenticate(ctx, request)
+
+		// early return ok if any of the sub-authenticators return ok
+		if err == nil {
+			return nil
 		}
 	}
-	return iface.NotAuthenticated()
+	return err
```

Comment:
We probably want to wrap this error. Right now this returns the last error, we want to either accumulate all the errors from the subauthenticators, or have a generic error about how non of the subauthenticators validated the request